addOns/reveal/src/test/java/org/zaproxy/zap/extension/reveal/ExtensionRevealUnitTest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2025 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.reveal;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import java.util.Locale;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.proxy.ProxyListener;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.utils.I18N;
+
+/** Unit test for {@link ExtensionReveal}. */
+class ExtensionRevealUnitTest {
+
+    private static final String DEFAULT_BODY =
+            "<html><head></head><body><H1>Some Heading</H1></body></html>";
+    private ExtensionReveal extension;
+
+    @BeforeEach
+    void setUp() {
+        Constant.messages = new I18N(Locale.ROOT);
+        extension = new ExtensionReveal();
+        extension.init();
+    }
+
+    @AfterEach
+    void cleanUp() {
+        Constant.messages = null;
+    }
+
+    @Test
+    void shouldAddProxyListenerAndApiImplementorOnHook() {
+        // Given
+        ExtensionHook extensionHook = mock(ExtensionHook.class);
+        // When
+        extension.hook(extensionHook);
+        // Then
+        ArgumentCaptor<ProxyListener> plArg = ArgumentCaptor.forClass(ProxyListener.class);
```

Comment:
Can also use `ExtensionReveal` since that's the `ProxyListener` that's being set.